flink-kubernetes/src/test/java/org/apache/flink/kubernetes/highavailability/KubernetesStateHandleStoreTest.java

```diff
@@ -1143,4 +1190,31 @@ private TestingLongStateHandleHelper.LongStateHandle addDeletingEntry(
         configMap.getData().put(key, deleting);
         return state;
     }
+
+    private CompletableFuture<Boolean> retryWithFirstFailedK8sOperation(
+            Function<KubernetesConfigMap, Optional<KubernetesConfigMap>> function,
+            KubernetesConfigMap leaderConfigMap) {
+        final AtomicInteger retryNum = new AtomicInteger(0);
+        final CompletableFuture<Boolean> result =
+                FutureUtils.retry(
+                        () ->
+                                CompletableFuture.supplyAsync(
+                                        () -> {
+                                            retryNum.incrementAndGet();
+                                            function.apply(leaderConfigMap);
+                                            if (retryNum.get() == 1) {
+                                                throw new KubernetesClientException(
+                                                        "unstable k8s client operation");
```

Comment:
```suggestion
                                                        "Expected exception to simulate unstable k8s client operation");
```
Can we make the exception method more explicit. This improves the readability of the logs produced during the test execution (i.e. it makes it easier to differentiate actual problematic Exceptions vs the once that are introduced as part of the certain tests).